src/SFA.DAS.EmployerIncentives.Api/Controllers/AccountQueryController.cs

```diff
@@ -57,5 +60,19 @@ public async Task<IActionResult> GetApplications(long accountId)
 
             return NotFound();
         }
+
+        [HttpGet("/legalentities/{hashedLegalEntityId}/employervendorid")]
```

Comment:
Wouldn't it be better and more RESTful to return a GetLegalEntityResponse?  Otherwise we are tying the API to a specific narrow use case.  If the use case changes then it would be a breaking change to the interface.

Could be "/legalentities/{hashedLegalEntityId}?employervendorid=123 to return the LegalEntity object for the vendorId.

Reply:
can be combined with existing GetLegalEntity in AccountQueryController? i.e. return LegalEntity based on the query string.

Reply:
Can't use that for the VRF status update function, only the hashed legal entity id is available to search on. The other endpoint required numeric account and account legal entity ids

Reply:
That's why it should be on the querystring (i.e. a search filter on the collection) rather than the path (which would be the Id of an individual resource)

Reply:
that's what i was thinking but then it probably won't look great with the if else or switch. perhaps we could pass in the query string to the query handler (somehow) ?

Reply:
Updated url